eclair-core/src/main/scala/fr/acinq/eclair/balance/BalanceActor.scala

```diff
@@ -75,33 +75,35 @@ private class BalanceActor(context: ActorContext[Command],
-              Monitoring.Metrics.BitcoinBalance.withTag(Monitoring.Tags.UtxoStatus, status).update(amount)
+              Monitoring.Metrics.BitcoinBalance.withTag(Monitoring.Tags.UtxoStatus, status).update(amount.toDouble)
           }
-
           previousBalance_opt match {
             case Some(previousBalance) =>
               log.info("on-chain diff={}", balance.onChain.total - previousBalance.onChain.total)
               val utxosBefore = previousBalance.onChain.confirmed ++ previousBalance.onChain.unconfirmed
               val utxosAfter = balance.onChain.confirmed ++ balance.onChain.unconfirmed
               val utxosAdded = utxosAfter -- utxosBefore.keys
               val utxosRemoved = utxosBefore -- utxosAfter.keys
-              utxosAdded.foreach { case (outPoint, amount) => log.info("+ utxo={} amount={}", outPoint, amount) }
-              utxosRemoved.foreach { case (outPoint, amount) => log.info("- utxo={} amount={}", outPoint, amount) }
+              utxosAdded.toList.sortBy(-_._2).foreach { case (outPoint, amount) => log.info("+ utxo={} amount={}", outPoint, amount) }
+              utxosRemoved.toList.sortBy(-_._2).foreach { case (outPoint, amount) => log.info("- utxo={} amount={}", outPoint, amount) }
 
               log.info("off-chain diff={}", balance.offChain.total - previousBalance.offChain.total)
               val offchainBalancesBefore = previousBalance.channels.view.mapValues(computeOffChainBalance(previousBalance.knownPreimages, _).total)
               val offchainBalancesAfter = balance.channels.view.mapValues(computeOffChainBalance(balance.knownPreimages, _).total)
-              offchainBalancesAfter
-                .map { case (channelId, balanceAfter) => (channelId, offchainBalancesBefore.getOrElse(channelId, Btc(0)), balanceAfter) }
-                .filter { case (_, balanceBefore, balanceAfter) => balanceAfter > balanceBefore }
-                .foreach { case (channelId, balanceBefore, balanceAfter) => log.info("+ channelId={} amount={}", channelId, balanceAfter - balanceBefore) }
-              offchainBalancesBefore
-                .map { case (channelId, balanceBefore) => (channelId, balanceBefore, offchainBalancesAfter.getOrElse(channelId, Btc(0))) }
-                .filter { case (_, balanceBefore, balanceAfter) => balanceBefore > balanceAfter }
-                .foreach { case (channelId, balanceBefore, balanceAfter) => log.info("- channelId={} amount={}", channelId, balanceBefore - balanceAfter) }
+              val offChainBalanceDiff = offchainBalancesAfter
+                .map { case (channelId, balanceAfter) => (channelId, balanceAfter - offchainBalancesBefore.getOrElse(channelId, Btc(0))) }
+                .toList
+              offChainBalanceDiff
+                .filter { case (_, diff) => diff > 0.sat }
+                .sortBy(-_._2)
+                .foreach { case (channelId, diff) => log.info("+ channelId={} amount={}", channelId, diff) }
+              offChainBalanceDiff
```

Comment:
nit: this is a bit different from what we had before: now there won't be an entry for utxos that are completely gone (since they wont be in `offchainBalancesAfter`). It probably does not matter since they are logged above ?

Reply:
Good catch! you're right that's not equivalent to what was before (and those are channel balances, not utxos, so the info is lost).

I attempted a fix in 89eb988df, it is much closer to had originally. WDYT?

Reply:
I can't find this commit ?

Reply:
Oops, pushed.